src/lib/builders/action-builder.ts

```diff
@@ -26,8 +26,11 @@ import { validate } from '../utils';
  */
 function actionBuildingFn(data: Specification.Action): () => Specification.Action {
   return () => {
-    validate('Action', data);
-    return data;
+    const result = {} as Specification.Action;
+
+    Object.assign(result, data);
+    validate('Action', result);
+    return result;
```

Comment:
thanks @JBBianchi 

I can change the property name to model, no problem 


The problem using constructor is that the code in the actual constructor is setting values we don't want in the final serialized json for example https://github.com/antmendoza/sdk-typescript/blob/from-types-to-classes/src/lib/definitions/function.ts#L18, we don't want type:rest if the client has not set it. Instead we want the value on deserialization.

Reply:
Indeed, and I don't really see how to prevent that... Maybe with a `prepareSerialization()` method or something that would need to be called on the whole object tree to mutate the serialized copy but that doesn't sound right... Or maybe using class-transformer for that? But that feels wrong as well :/